matter/wifi/rs911x/rsi_if.c

```diff
@@ -492,35 +495,36 @@ void wfx_rsi_task(void *arg)
         wfx_wifi_scan_result_t ap;
         rsi_scan_info_t *scan;
         int32_t status;
-        uint8_t bgscan_results[500] = { 0 };
-        status                      = rsi_wlan_bgscan_profile(1, (rsi_rsp_scan_t *)bgscan_results, 500);
+        uint8_t bgscan_results[BG_SCAN_RES_SIZE] = {0};
+        status =
+            rsi_wlan_bgscan_profile(1, (rsi_rsp_scan_t *)bgscan_results, BG_SCAN_RES_SIZE);
 
-        WFX_RSI_LOG("%s: status: %02x size = %d", __func__, status, 500);
+        WFX_RSI_LOG("%s: status: %02x size = %d", __func__, status, BG_SCAN_RES_SIZE);
         rsi_rsp_scan_t *rsp = (rsi_rsp_scan_t *)bgscan_results;
         if (status) {
           /*
-         * Scan is done - failed
-         */
+           * Scan is done - failed
+           */
         } else
           for (x = 0; x < rsp->scan_count[0]; x++) {
             scan = &rsp->scan_info[x];
             strcpy(&ap.ssid[0], (char *)&scan->ssid[0]);
             if (wfx_rsi.scan_ssid) {
               WFX_RSI_LOG("Inside scan_ssid");
-              WFX_RSI_LOG("SCAN SSID: %s , ap scan: %s",wfx_rsi.scan_ssid,ap.ssid);
-              if(strcmp(wfx_rsi.scan_ssid,ap.ssid) == 0){
+              WFX_RSI_LOG("SCAN SSID: %s , ap scan: %s", wfx_rsi.scan_ssid,
+                          ap.ssid);
+              if (strcmp(wfx_rsi.scan_ssid, ap.ssid) == CMP_SUCCESS) {
                 WFX_RSI_LOG("Inside ap details");
                 ap.security = scan->security_mode;
-                ap.rssi     = (-1) * scan->rssi_val;
-                memcpy(&ap.bssid[0], &scan->bssid[0], 6);
+                ap.rssi = (-1) * scan->rssi_val;
+                memcpy(&ap.bssid[0], &scan->bssid[0], COPY_6_CHAR);
```

Comment:
Think of my suggested define earlier for BSSID

Reply:
changed